resources/eventing/charts/controller/templates/oauth2-client.yaml

```diff
@@ -4,9 +4,12 @@ kind: OAuth2Client
 metadata:
   name: {{ include "controller.fullname" . }}
   labels: {{- include "controller.labels" . | nindent 4 }}
+  finalizers:
```

Comment:
I suggest to never ever add finalisers coming from a controller since it will block deletion of a resource if the controller is gone. Rather the edited resource should be patched via strategic merge, ignoring the finalisers. I suggest to review the logic that forces this field to be present

Reply:
Would it be possible then to configure reconciler not to remove the `finalizers` if they were applied by hydra-maester controller? For the sake of fixing the error that would also be OK

Reply:
😕 reconciler removes finalizers? In what scenario?

Reply:
TBH also surprising for me. Finalisers outside of the reconciler should be ignored imo. Also setting nil metadata would result in them being removed on every reconcile. @tobiscr are you aware of the behaviour here? I almost cannot imagine that this never was an issue before...

Reply:
The finalisers are ignored by the old reconciler until we are deleting the cluster. During the deletion, one of the final steps is the cleanup of finalisers to make sure every resources gets dropped (just to make sure that there are no "dead bodies" left).

Reply:
If you have to deal with finalisers during runtime, you can either handle such cases in a custom reconciler (but this will update finalisers only during the regular reconciliation runs) or you handle it over an operator during runtime.

Reply:
Removed the finalizers as those will be handled by https://github.com/kyma-incubator/reconciler/pull/1274